native/src/main/java/io/ballerina/stdlib/grpc/logging/GrpcLogManager.java

```diff
@@ -0,0 +1,168 @@
+import static io.ballerina.stdlib.grpc.GrpcConstants.GRPC_TRACE_LOG_HOST;
+import static io.ballerina.stdlib.grpc.GrpcConstants.GRPC_TRACE_LOG_PORT;
+import static io.ballerina.stdlib.grpc.GrpcConstants.HTTP_ACCESS_LOG;
+import static io.ballerina.stdlib.grpc.GrpcConstants.HTTP_ACCESS_LOG_ENABLED;
+import static io.ballerina.stdlib.grpc.GrpcConstants.HTTP_TRACE_LOG;
+import static io.ballerina.stdlib.grpc.GrpcConstants.HTTP_TRACE_LOG_ENABLED;
+
+/**
+ * Java util logging manager for ballerina which overrides the readConfiguration method to replace placeholders
+ * having system or environment variables.
+ */
+public class GrpcLogManager extends LogManager {
+
+    static {
+        // loads logging.properties from the classpath
+        try (InputStream is = GrpcLogManager.class.getClassLoader().
+                getResourceAsStream("logging.properties")) {
+            LogManager.getLogManager().readConfiguration(is);
+        } catch (IOException e) {
+            throw new RuntimeException("failed to read logging.properties file from the classpath", e);
+        }
+    }
+
+    protected Logger grpcTraceLogger;
+    protected Logger grpcAccessLogger;
+
+    public GrpcLogManager(boolean traceLogConsole, BMap traceLogAdvancedConfig, BMap accessLogConfig) {
+
+        this.setHttpTraceLogHandler(traceLogConsole, traceLogAdvancedConfig);
+        this.setHttpAccessLogHandler(accessLogConfig);
+    }
+
+    /**
+     * Initializes the GRPC trace logger.
+     */
+    public void setHttpTraceLogHandler(boolean traceLogConsole, BMap traceLogAdvancedConfig) {
+
+        if (grpcTraceLogger == null) {
+            // keep a reference to prevent this logger from being garbage collected
+            grpcTraceLogger = Logger.getLogger(HTTP_TRACE_LOG);
+        }
+        PrintStream stdErr = System.err;
+        boolean traceLogsEnabled = false;
+
+        Boolean consoleLogEnabled = traceLogAdvancedConfig.getBooleanValue(GRPC_LOG_CONSOLE);
+        if (traceLogConsole || consoleLogEnabled) {
+            ConsoleHandler consoleHandler = new ConsoleHandler();
+            consoleHandler.setFormatter(new GrpcTraceLogFormatter());
+            consoleHandler.setLevel(Level.FINEST);
+            grpcTraceLogger.addHandler(consoleHandler);
+            traceLogsEnabled = true;
+        }
+
+        BString logFilePath = traceLogAdvancedConfig.getStringValue(GRPC_LOG_FILE_PATH);
+        if (logFilePath != null && !logFilePath.getValue().trim().isEmpty()) {
+            try {
+                FileHandler fileHandler = new FileHandler(logFilePath.getValue(), true);
+                fileHandler.setFormatter(new GrpcTraceLogFormatter());
+                fileHandler.setLevel(Level.FINEST);
+                grpcTraceLogger.addHandler(fileHandler);
+                traceLogsEnabled = true;
+            } catch (IOException e) {
+                throw new RuntimeException("failed to setup GRPC trace log file: " + logFilePath.getValue(), e);
```

Comment:
```suggestion
                throw new RuntimeException("Failed to setup gRPC trace log file: " + logFilePath.getValue(), e);
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/413/commits/303299de9612deb38959f15b7389ed952825454a